packages/react-button/src/components/Button/useButton.ts

```diff
@@ -1,28 +1,34 @@
 import * as React from 'react';
-import { getStyleFromPropsAndOptions } from '@fluentui/react-theme-provider';
-import { useFocusRects } from '@uifabric/utilities';
-import { ComposePreparedOptions } from '@fluentui/react-compose';
-import { ButtonProps, ButtonState } from './Button.types';
+import { ButtonState } from './Button.types';
+// import { useMergedRefs } from '@uifabric/react-hooks';
 
 /**
- * The useButton hook processes the Button component props and returns state.
- * @param props - Button props to derive state from.
+ * The useButton hook processes the Button draft state.
+ * @param draftState - Button draft state to mutate.
  */
-export const useButton = (
-  props: ButtonProps,
-  ref: React.Ref<HTMLElement>,
-  options: ComposePreparedOptions,
-): ButtonState => {
-  const buttonRef = React.useRef<HTMLButtonElement | null>(null);
+export const useButton = (draftState: ButtonState) => {
+  // Update the button's tab-index, keyboard handling, and aria attributes.
+  if (draftState.as !== 'button') {
+    draftState.role = 'button';
+    draftState.tabIndex = 0;
 
-  React.useImperativeHandle(props.componentRef, () => ({
-    focus: () => buttonRef.current?.focus(),
-  }));
-  useFocusRects(buttonRef);
+    if (draftState.as !== 'a') {
+      const { onKeyDown, onClick } = draftState;
 
-  return {
-    ...props,
-    buttonRef,
-    style: getStyleFromPropsAndOptions(props, options, '--button'),
-  };
+      draftState['data-isFocusable'] = true;
+
+      draftState.onKeyDown = (ev: React.KeyboardEvent<HTMLElement>) => {
+        if (onKeyDown) {
+          onKeyDown(ev);
+        }
+
+        if (!ev.defaultPrevented && onClick && (ev.which === 20 || ev.which === 13)) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
It means that typings are not correct and it can be surprise for a client to receive `React.KeyboardEvent` instead of `React.MouseEvent`

Reply:
I've resolved this by calling `click` to force a `MouseEvent`.